DOCSP-37089-new Install with Docker task page (#101)
mongodb/docs-relational-migrator: source/installation/install-docker.txt

         select the appropriate Relational Migrator installer.

      #. Create a docker image using a Relational Migrator installer.

         For a complete example, you can download the following docker-compose: 
         `Containerised kafka-connect cluster with Relational Migrator 
         <https://migrator-installer-repository.s3-ap-southeast-2.amazonaws.com/1.4.3/docker-compose-migrator-single-kafka.yml>`__.

   .. step:: If your source database is MySQL or Oracle, add their related drivers

      Relational Migrator uses the JDBC driver of the respective source database to read 
      database schema. It bundles SQL Server and PostgreSQL JDBC drivers. For MySQL and 
      Oracle, you must add their drivers to the following path:

      ``<path-to-installation>/mongodb-relational-migrator/lib/app/lib``
      
      - Oracle: `21.6.0.0 of odjbc11.jar from the Oracle 21c
        <https://www.oracle.com/database/technologies/appdev/jdbc-drivers-archive.html>`__
      - MySQL: `latest 8.0.x version, Platform Independent Version 
        <https://dev.mysql.com/downloads/connector/j/>`__

Learn More
----------

> - :ref:`kafka-intro`

Author: [For reviewer]

These currently cause build errors but they will be resolved once https://github.com/mongodb/docs-relational-migrator/pull/96 is merged.

Before you Begin
----------------

You must have Docker installed on your computer. For more information, see  
`Install Docker Engine <https://docs.docker.com/engine/install/>`__. 

Steps
-----

.. procedure::
   :style: normal

   .. step:: Install a Kafka cluster

      Use the following ``docker-compose`` configuration as an example to install a new 
      Kafka cluster, or connect to an existing cluster.

      .. code-block::

         version: '3'
         services:
           kafka-1:
           image: quay.io/debezium/kafka:2.2
           ports:
             - 19092:9092
             - 19093:9093
           environment:
             CLUSTER_ID: oh-sxaDRTcyAr6pFRbXyzA
             BROKER_ID: 1
             KAFKA_CONTROLLER_QUORUM_VOTERS: 1@kafka-1:9093
             NODE_ROLE: controller,broker
           healthcheck:
             test: ["CMD-SHELL", "./bin/kafka-topics.sh --version"]
             interval: 5s
             timeout: 10s
             retries: 10

   .. step:: Install a Kafka Connect cluster

      Install a Kafka Connect cluster with the ``relational-migrator-connect`` image 

>       from the `public Amazon ECR <public.ecr.aws/v4d7k6c9/relational-migrator:latest>`__.

Reviewer: This is should not be a hyperlink; we need to show the full path of the container in the docs.

Reviewer 2: This link does not work for me

   .. step:: Install Relational Migrator

      You have two options for installing Relational Migrator:

      1. From the `download center <https://www.mongodb.com/try/download/relational-migrator>`__, 
         select the appropriate Relational Migrator installer.

      #. Create a docker image using a Relational Migrator installer.

         For a complete example, you can download the following ``docker-compose``: 
         `Containerised kafka-connect cluster with Relational Migrator 

>          <https://migrator-installer-repository.s3-ap-southeast-2.amazonaws.com/1.4.3/docker-compose-migrator-single-kafka.yml>`__.

Reviewer: This single Docker Compose file is probably the most important thing to include in the docs as it shows everything in one place. I'd suggest including this directly in the docs (we probably don't need the smaller compose fragments).
The download hyperlink is nice but it's tied to a specific (now out-of-date) version. @alwaysbemark can we publish a version of this with a stable URL that pulls the `:latest` tag?

Author: Updated to include the example. I left the current link at the moment but will update once a stable URL is available.

Reviewer 3: We have a downloads MongoDB mirror now so we should use https://downloads.mongodb.org/migrator/1.5.0/docker-compose-migrator-single-kafka.yml here.
It probably makes sense to always have it versioned as things could change that could break the Docker compose file compatibility.

Reviewer: @alwaysbemark It's fine to version the compose files, but we don't want to have to bake a version number into the docs as it will get out of date quickly; is there a stable URL for the latest version? If not, we should just link to the main download page (https://www.mongodb.com/try/download/relational-migrator) and make the compose files downloadable from the list.

.. _install-docker:

=================================================
Install with a Containerized Cluster Using Docker
=================================================

.. default-domain:: mongodb

.. facet::
   :name: genre
   :values: tutorial

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

You can install Relational Migrator with Kafka in a containerized environment 
using `Docker <https://docs.docker.com/engine/>`__. This method configures Kafka 
to store data locally in Docker containers. This deployment method is ideal for 

> educational purposes. Use this if you are looking to learn and understand how to 

Reviewer 4: Suggestion for brevity:

```suggestion
educational purposes. Use this if you want to learn how to 
```

Author: Updated.

configure your own multi-server Kafka environment. 

> .. note::

Reviewer 4: Maybe escalate to warning?

Author: Updated.

   This deployment method is not recommended for production workloads. 

About this Task
---------------

- Relational Migrator is available port 8080.

- In the Docker container, Relational Migrator is installed to 

>   ``/opt/mongodb-relational-migrator/``.

Reviewer 4: These look like "Before you Begin" requirements

Author: These are actually not prereq's but more of general information so I will go ahead and leave this under About this Task for now.

- When running in Docker, the ``user.properties`` file is ignored. Define any 
  property overrides using environment variables. For example: 

  .. code-block::

     docker run -p 8080:8080 -e MIGRATOR_APP_TELEMETRY_ENABLE=false public.ecr.aws/v4d7k6c9/relational-migrator:latest

> - `Docker volumes <https://docs.docker.com/storage/volumes/>`__ can be used for 

Reviewer 4: Suggest taking these out of list format and into normal paragraphs

Author: Thank you but just for now I might keep them in a list due to the fact that they are all kind of varying on subjects but pending any removal of bullet points I will go ahead and place them into paragraphs.

Reviewer 4: That makes sense if you're keeping the other info!

  persisting data and/or drivers if a container stops running or is restarted. 
  You can create volumes to the paths:

  - For data and logs: ``/root/Migrator``
  - For drivers: ``/opt/mongodb-relational-migrator/lib/app/lib``

  The following example runs Relational Migrator with persistent volumes for drivers 
  and data:

  .. code-block::

     docker run -p 8080:8080 -v /migrator/drivers:/opt/mongodb-relational-migrator/lib/app/lib -v\
     /migrator/data:/root/Migrator  public.ecr.aws/v4d7k6c9/relational-migrator:latest

Before you Begin
----------------

You must have Docker installed on your computer. For more information, see  
`Install Docker Engine <https://docs.docker.com/engine/install/>`__. 

Steps
-----

.. procedure::
   :style: normal

   .. step:: Install a Kafka cluster

>       Use the following ``docker-compose`` example to install a Kafka cluster.
>       
>       You can also use an existing cluster and provide configuration details to connect 
>       your existing cluster.

Reviewer 4: Suggestion for simplicity:

```suggestion
      Use the following ``docker-compose`` configuration as an example to install a new 
      Kafka cluster, or connect to an existing cluster.
```

Author: Updated.

      .. code-block::

         version: '3'
         services:
           kafka-1:
           image: quay.io/debezium/kafka:2.2
           ports:
             - 19092:9092
             - 19093:9093
           environment:
             CLUSTER_ID: oh-sxaDRTcyAr6pFRbXyzA
             BROKER_ID: 1
             KAFKA_CONTROLLER_QUORUM_VOTERS: 1@kafka-1:9093
             NODE_ROLE: controller,broker
           healthcheck:
             test: ["CMD-SHELL", "./bin/kafka-topics.sh --version"]
             interval: 5s
             timeout: 10s
             retries: 10

   .. step:: Install a Kafka Connect cluster

      Install a Kafka Connect cluster with the ``relational-migrator-connect`` image 

>       from `public Amazon ECR <public.ecr.aws/v4d7k6c9/relational-migrator:latest>`__.

Reviewer 4: ```suggestion
      from the `public Amazon ECR <public.ecr.aws/v4d7k6c9/relational-migrator:latest>`__.
```

Author: Updated.
